### PR TITLE
Refactor rvine structure

### PR DESCRIPTION
**Pull request by tvatter**

### CODE REVIEW

**Comment by tvatter** [comment]
Oh boy, checks are failing on windows only....

File: include/vinecopulib/misc/tools_serialization.hpp [hunk 1]
```diff
@@ -41,13 +41,13 @@ inline boost::property_tree::ptree matrix_to_ptree(
...
-//! conversion from vinecopulib::RVineMatrix to boost::property_tree::ptree
+//! conversion from vinecopulib::TriangularArray to boost::property_tree::ptree
 //!
-//! @param matrix the vinecopulib::RVineMatrix to convert.
+//! @param matrix the vinecopulib::TriangularArray to convert.
 //! @return the corresponding boost::property_tree::ptree.
 template<class T>
-inline boost::property_tree::ptree rvinematrix_to_ptree(
-    RVineMatrix<T> matrix)
+inline boost::property_tree::ptree TriangularArray_to_ptree(
```
Comment (tvatter) [inline include/vinecopulib/misc/tools_serialization.hpp]: We never used CamelCase in functions names I think.

File: include/vinecopulib/misc/tools_serialization.hpp [hunk 2]
```diff
@@ -124,12 +124,12 @@ inline Eigen::Matrix <T, Eigen::Dynamic, Eigen::Dynamic> ptree_to_matrix(
...
 
-//! conversion from boost::property_tree::ptree to vinecopulib::RVineMatrix
+//! conversion from boost::property_tree::ptree to vinecopulib::TriangularArray
 //!
 //! @param iroot the boost::property_tree::ptree to convert.
-//! @return the corresponding vinecopulib::RVineMatrix
+//! @return the corresponding vinecopulib::TriangularArray
 template<typename T>
-inline RVineMatrix<T> ptree_to_rvinematrix(
+inline TriangularArray<T> ptree_to_TriangularArray(
```
Comment (tvatter) [inline include/vinecopulib/misc/tools_serialization.hpp]: same comment as above

File: test/src_test/include/test_rvine_structure.hpp [hunk 3]
```diff
@@ -16,7 +16,7 @@ using namespace vinecopulib;
 
 TEST(rvine_structure, rvine_matrix_works) {
```
Comment (tvatter) [inline test/src_test/include/test_rvine_structure.hpp]: triangular_array_works

File: include/vinecopulib/vinecop/rvine_structure.hpp [hunk 4]
```diff
@@ -11,6 +11,56 @@
 
 namespace vinecopulib {
 
+//! R-vine structures
```
Comment (tvatter) [inline include/vinecopulib/vinecop/rvine_structure.hpp]: Should we mention how it is stored (i.e., as an order vector and the upper-triangle in natural order)?

File: test/src_test/include/test_rvine_structure.hpp [hunk 5]
```diff
@@ -63,7 +63,7 @@ using namespace vinecopulib;
 //    timer.end();
 //
 //    timer.start();
-//    RVineMatrix2 rvm(mat, false);
+//    TriangularArray2 rvm(mat, false);
 //    timer.end();
 //
 //    std::cout << "matrix ------" << std::endl;
```
[the remark this answers is not in the document]
Comment (tnagler) [reply test/src_test/include/test_rvine_structure.hpp]: I think you reviewed an old commit?

File: test/src_test/include/test_rvine_structure.hpp (hunk 3 above)
Comment (tnagler) [reply test/src_test/include/test_rvine_structure.hpp]: yes!

File: include/vinecopulib/vinecop/rvine_structure.hpp (hunk 4 above)
Comment (tnagler) [reply include/vinecopulib/vinecop/rvine_structure.hpp]: I mention this at the end of the class docs.

Comment (tvatter) [reply include/vinecopulib/vinecop/rvine_structure.hpp]: OK my bad didn't read properly

File: test/src_test/include/test_rvine_structure.hpp (hunk 5 above)
[the remark this answers is not in the document]
Comment (tvatter) [reply test/src_test/include/test_rvine_structure.hpp]: Indeed

File: include/vinecopulib/misc/tools_serialization.hpp (hunk 1 above)
Comment (tvatter) [reply include/vinecopulib/misc/tools_serialization.hpp]: Yeah I figured, I always use "match case" to avoid this ;)

**Pull request merged**